### DISCUSSION TITLE
Recommended way to control optional values

### DISCUSSION THREAD

**Discussion by wsaca** [Netflix/dgs-framework#739] [Q&A]
I'm currently using the next class to control undefined params in mutations:
```java
public class UpdateApiInput {
    private String apiId;
    private Undefined<String> name = Undefined.empty();

    public void setName(String name) {
        this.name = Undefined.of(name);
    }
}
````

But after version 4.5.1 this class is not working because the code changed to use the new [InputObjectMapper](https://github.com/Netflix/dgs-framework/blob/master/graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/InputObjectMapper.kt) instead of jackson ObjectMapper:
- Commit: https://github.com/Netflix/dgs-framework/commit/2fe40a9fd1acab3037354586338dcede7c4ea9b7

Now my test are failing with the next error because the method setName(...) is ignored:
```java
com.netflix.graphql.dgs.exceptions.DgsInvalidInputArgumentException: Invalid input argument `new-name` for field `name` on type `x.graphql.mutation.input.UpdateApiInput`
	at com.netflix.graphql.dgs.internal.InputObjectMapper.trySetField(InputObjectMapper.kt:152)
	at com.netflix.graphql.dgs.internal.InputObjectMapper.mapToJavaObject(InputObjectMapper.kt:127)
	at com.netflix.graphql.dgs.internal.DataFetcherInvoker.convertValue(DataFetcherInvoker.kt:241)
```

Do you know how I could resolve my problem with the current version, maybe you have a different solution.

**Comment by paulbakker** [comment]
That's not going to work anymore unfortunately. This implicitly worked previously, but wasn't really a supported use case.

We _could_ add support for `Optional`, which would be similar. I'm not sure what `Undefined` is, but I'm guessing that's some other library?

**Reply by wsaca** [reply]
There is a discussion about my case here https://github.com/spring-projects/spring-graphql/pull/140. The alternative is to use DataFetchingEnvironment, but it adds extra code.
